app/src/main/java/com/github/sdpcoachme/database/CachingStore.kt

```diff
@@ -487,58 +469,56 @@ class CachingStore(private val wrappedDatabase: Database,
     fun getSchedule(currentWeekMonday: LocalDate): CompletableFuture<Schedule> {
         currentShownMonday = currentWeekMonday
 
-        return CompletableFuture.allOf(registerForGroupEventFuture, groupEventFuture).thenCompose {
-            getCurrentEmail().thenCompose { email ->
-                if (cachedSchedule.events.isEmpty() && cachedSchedule.groupEvents.isEmpty()) {  // If no cached schedule for that account, we fetch the schedule from the db
-                    wrappedDatabase.getSchedule(email).thenApply { schedule ->
-                        val events =
-                            schedule.events.filter {   // We only cache the events that are in the current week or close to it
-                                if (it != Event()) {
-                                    val start = LocalDateTime.parse(it.start).toLocalDate()
-                                    val end = LocalDateTime.parse(it.end).toLocalDate()
-                                    start >= minCachedMonday && end <= maxCachedMonday
-                                } else {
-                                    false
-                                }
+        return getCurrentEmail().thenCompose { email ->
+            if (cachedSchedule.events.isEmpty() && cachedSchedule.groupEvents.isEmpty()) {  // If no cached schedule for that account, we fetch the schedule from the db
+                wrappedDatabase.getSchedule(email).thenApply { schedule ->
+                    val events =
+                        schedule.events.filter {   // We only cache the events that are in the current week or close to it
+                            if (it != Event()) {
```

Comment:
It would be nice to refactor the code to remove code block duplication but, due to time constraints, I believe it is best to move this to a later time and merge this pr.